buildkite/bazel-central-registry/bcr_compatibility.py

```diff
@@ -0,0 +1,140 @@
+#!/usr/bin/env python3
+#
+# Copyright 2024 The Bazel Authors. All rights reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#    http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+# pylint: disable=line-too-long
+# pylint: disable=missing-function-docstring
+# pylint: disable=unspecified-encoding
+# pylint: disable=invalid-name
+"""The CI script for BCR Bazel Compatibility Test pipeline."""
+
+
+import os
+import sys
+import subprocess
+
+import bazelci
+import bcr_presubmit
+from bcr_presubmit import BcrPipelineException
+
+CI_MACHINE_NUM = {
```

Comment:
hm this is a bit dangerous since it can get out of sync with reality. For non-Mac machines we have the canonical numbers in https://github.com/bazelbuild/continuous-integration/blob/master/buildkite/instances.yml

Reply:
Yeah, but those non-Mac numbers basically never change, right? I don't want to curl another yml file for this, and I'll keep it updated when our Mac fleet expands.

Reply:
ok, fine with me.